TokensModal only show native token when possible

kev1n-peters opened the pull request:
Prevent users from receiving wrapped tokens when they could receive the  native version.

artursapek commented on wormhole-connect/src/components/TokensModal.tsx:
```diff
@@ -511,6 +511,16 @@ function TokensModal(props: Props) {
+    if (type === 'dest') {
+      const nativeToken = supportedTokens.find((t) => t.nativeChain === chain);
+      if (nativeToken) {
+        filtered = [nativeToken];
```
is it possible there's multiple native tokens eventually? like if Connect can perform swaps?

this logic is basically "find the first native token and if it exists, only offer that." we could also have it be "find all the native tokens, if any exist, offer them all."

kev1n-peters replied:
I think we can change it to be "future proof"

kev1n-peters replied:
changed

artursapek commented on wormhole-connect/src/components/TokensModal.tsx:
```diff
@@ -511,6 +511,16 @@ function TokensModal(props: Props) {
+    // This is to avoid users inadvertently receiving wrapped versions of the token.
+    if (type === 'dest') {
+      const nativeTokens = filtered.filter((t) => t.nativeChain === chain);
+      if (nativeTokens) {
```
```suggestion
      if (nativeTokens.length > 0) {
```

an empty array is truthy

kev1n-peters replied:
fixed

kev1n-peters committed:
TokensModal only show native token when possible

Prevent users from receiving wrapped tokens when they could receive
the native version.
